modules/core/src/main/java/org/apache/ignite/internal/processors/performancestatistics/PerformanceStatisticsProcessor.java

```diff
@@ -278,6 +325,35 @@ private void stopWriter() {
         log.info("Performance statistics writer stopped.");
     }
 
+    /** Rotate performance statistics writer. */
+    private void rotateWriter() throws IgniteCheckedException, IOException {
+        try {
+            FilePerformanceStatisticsWriter oldWriter = null;
+
+            synchronized (mux) {
+                if (writer != null) {
+                    FilePerformanceStatisticsWriter newWriter = new FilePerformanceStatisticsWriter(ctx);
+
+                    newWriter.start();
+
+                    oldWriter = writer;
+
+                    writer = newWriter;
+
+                    oldWriter.stop();
+                }
+            }
+
+            if (oldWriter != null)
+                log.info("Performance statistics writer rotated[writtenFile=" + oldWriter.getFile() + "].");
+        }
+        catch (IgniteCheckedException | IOException e) {
+            log.error("Failed to rotate performance statistics writer.", e);
```

Comment:
Why do we need to log exception here? 
Is it not logged elsewhere?

Reply:
we will get an error when getting the result.
I find this information useful here